### PR TITLE
Assets MR

### PR DESCRIPTION
**Pull request by mohitDehuliya**

### CODE REVIEW

File: repository/asset_repository.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package repository
+
+import (
+	"github.com/jmoiron/sqlx"
+)
+
+type AssetRepository interface {
```
Comment (vkhichar) [inline repository/asset_repository.go:9]: Add your method here

File: repository/asset_repository.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package repository
+
+import (
+	"context"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type AssetRepository interface {
+	CreateAsset(ctx context.Context, assetCategory string, assetStatus string, purchaseCost float64, assetName string, spec *domain.Specs) (*domain.Asset, error)
```
Comment (vkhichar) [inline repository/asset_repository.go:10]: input argument should be domin.Asset

File: service/asset_service.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package service
+
+import (
+	"context"
+
+	"github.com/vkhichar/asset-management/repository"
+)
+
+type AssetService interface {
+	CreateAsset(ctx context.Context, assetCategory string, assetStatus string, purchaseCost float64, assetName string, spec *domain.Specs) (*domain.Asset, error)
```
Comment (vkhichar) [inline service/asset_service.go:10]: Input argument should be domain.Asset

File: domain/assets.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	"time"
+)
+
+type UUID [16]byte
+
+type assets struct {
+	Id             UUID                   `db:"id"` //uuid package read
+	Status         string                 `db:"status"`
+	Category       string                 `db:"category"`
+	Purchaseat     time.Time              `db:"purchase_at"`
```
Comment (vkhichar) [inline domain/assets.go:13]: variable name should be PurchaseAt

File: domain/assets.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	Purchasecost   float64                `db:"purchase_cost"`
```
Comment (vkhichar) [inline domain/assets.go:14]: variable name should be PurchaseCost

File: domain/assets.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+	Name           string                 `db:"name"`
+	Specifications map[string]interface{} `db:"specifiation"` //encoding json read
```
Comment (vkhichar) [inline domain/assets.go:16]: whats the key name for 'Specifications' in db migrations?

change the db key accordingly

File: domain/assets.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package domain
+
+import (
+	"time"
+)
+
+type UUID [16]byte
+
+type assets struct {
```
Comment (ranbharerp15) [inline domain/assets.go:9]: Rename struct as Asset. The domain struct name should be noun in singular form.

File: service/asset_service.go [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+type assetService struct {
```
Comment (ranbharerp15) [inline service/asset_service.go:13]: @vkhichar We can name implementing struct as AssetServiceImpl right?

File: handler/routes.go [hunk 9]
```diff
@@ -5,4 +5,5 @@ import "net/http"
 func Routes() {
 	http.Handle("/ping", PingHandler())
 	http.Handle("/login", LoginHandler(deps.userService))
+	http.Handle("/assets", ListAssetHandler(deps.assetService))
```
Comment (ranbharerp15) [inline handler/routes.go:8]: Rename the func as ListAssetsHandler. 
Please check if asset handler file is also pushed here.

File: repository/asset_repository.go [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+type assetRepo struct {
```
Comment (ranbharerp15) [inline repository/asset_repository.go:13]: @vkhichar Can we name implementing struct as AssetRepositoryImpl?

**Review by ranbharerp15 — CHANGES_REQUESTED** [review]

File: service/asset_service.go (hunk 8 above)
Comment (vkhichar) [reply service/asset_service.go:13]: no need to add `Impl` suffix, It's understood that it's an implementation, without putting the suffix too
though, there is no standard for this naming convention

**Pull request closed without merging**